src/amp/components/Experiment.tsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+import { AmpModelCollection } from '@root/src/amp/lib/experiment';
```

Comment:
what's in `AmpModelCollection`? i'm not quite sure from the name?

what if it was just `AmpExperiment`?

Reply:
```typescript
interface ExperimentCollection<ExperimentObject> {
    [experimentName: string]: ExperimentObject;
}

export type AmpModelCollection = ExperimentCollection<AmpModel>;

export type StyledModelCollection = ExperimentCollection<StyledExperimentModel>;
```
`AmpModelCollection` and `StyledModelCollection` are objects containing data for each AB Test we'd be running. Google requires that we add the data to the `amp-experiment` tag in the format I've called `AmpModelCollection`, which is basically `{ExperimentName: ExperimentConfig}`
In a bid to try and keep the styles associated with a given experiment and variant close to the definition of the variant, `StyledModelCollection` puts the CSS string for the variant style as a property of the `variant` field in the `StyledModel`.

The data model I've ended up with here is arguably way too complex given that it started as an attempt to keep things simple. Ideally we'd have something like the following.
## Data Model ##
### Experiment Definition ###
```json
{
      "button-color-experiment": {
"sticky": false,
        "variants": {
          "0": 30,
          "1": 30,
          "2": 30
        }
      }
    }
```
### Style definition ###
```json
{
      "button-color-experiment": {
          "0": "style1",
          "1": "style2",
          "2": "style3"
        }
    }
```
### Overall definition of an experiment ###
 Combination of the above in something tidy and not confusing...

I'm not sure the best way to define that data model without it getting inevitably complex.